examples/platforms/simulation/radio.c

```diff
@@ -996,7 +1059,14 @@ void radioProcessFrame(otInstance *aInstance)
 
 exit:
 
+#if OPENTHREAD_SIMULATION_EXT_RF_MODELS
+    // If Rx-frame needs to be ACKed, postpone the receive-done report until the ACK is sent, but only if the frame-rx
+    // went ok. This prevents the device sending more frames before the ACK has a chance to be sent.
```

Comment:
'isAcked' here tracks whether the read frame requires an Ack to be sent, or not. When no Ack is required, the logic is identical to the original code: it will report otPlatRadioReceiveDone() to handle the event of Rx done, if the frame was indeed destined to this device.

When an Ack is required, the handler otPlatRadioReceiveDone() is not called immediately - it is postponed. First the Ack is scheduled, and later on when the simulator sends the "Tx Done" event for this Ack then in the handler of that event it is checked whether the radio is actually in "receiving" state (if yes, in this case it means an Ack was either successfully transmitted or was attempted to be transmitted but failed). If that is the case, then the handler is called:
   otPlatRadioReceiveDone(aInstance, &sReceiveFrame, OT_ERROR_NONE);

The idea was that the sReceiveFrame of the last received frame is still sitting in the frame buffer and wasn't changed in the meantime.  (That assumption seems to hold due to the short time between frame reception and start of Ack sending, AIFS = 192 us, which is shorter than any PHY frame would be. Still a bit risky maybe.)

The reason this was done is that as soon as the higher-layer handler is called for a received frame, the OT stack may respond with any kind of new message (MLE, CoAP, ...) immediately and this may cause the (sub)MAC layer to switch from 'receive' to 'transmit' state and try doing CSMA, etc., which shouldn't be allowed because the Ack is still waiting to be transmitted.

Maybe better is we introduce a state variable "sRxWait" to track this?  So that the higher subMAC/MAC layers cannot grab the radio for transmission/CSMA when the Ack is still pending to be transmitted.